File: benchmarks/poly/jsonserialize/jsonserialize.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
	"time"
)

func main() {
	t0 := time.Now()
	parts := make([]string, 0, 400000)
	for i := 0; i < 400000; i++ {
		s := strconv.Itoa(i)
		parts = append(parts, `{"id":`+s+`,"name":"user`+s+`","score":`+strconv.Itoa(i%100)+`}`)
	}
	out := strings.Join(parts, "\n")
	fmt.Println(len(out))
	fmt.Println(len(parts))
	fmt.Fprintf(os.Stderr, "bench_ns=%d\n", time.Since(t0).Nanoseconds())
}

File: benchmarks/poly/jsonserialize/jsonserialize.rs
fn main() {
    let t0 = std::time::Instant::now();
    let mut parts: Vec<String> = Vec::with_capacity(400_000);
    for i in 0..400_000 {
        parts.push(format!(
            "{{\"id\":{},\"name\":\"user{}\",\"score\":{}}}",
            i,
            i,
            i % 100
        ));
    }
    let out = parts.join("\n");
    println!("{}", out.len());
    println!("{}", parts.len());
    eprintln!("bench_ns={}", t0.elapsed().as_nanos());
}
